src/Transports.Subscriptions.Abstractions/SubscriptionManager.cs

```diff
@@ -82,13 +89,30 @@ public Task UnsubscribeAsync(string id)
                 payload.OperationName,
                 payload.Query);
 
-            var result = await _executer.ExecuteAsync(
-                payload.OperationName,
-                payload.Query,
-                payload.Variables?.ToInputs(),
-                context,
-                null // TODO: find later a better way to specify services
-            ).ConfigureAwait(false);
+            ExecutionResult result;
+            if (_serviceScopeFactory != null)
```

Comment:
Well, maybe to add something like that
```c#
internal sealed class NoopFactory : IServiceScopeFactory
        {
            public static readonly NoopFactory Instance = new NoopFactory();

            public IServiceScope CreateScope() => NoopServiceScope.Instance;

            internal sealed class NoopServiceScope : IServiceScope
            {
                public static readonly NoopServiceScope Instance = new NoopServiceScope();

                public IServiceProvider ServiceProvider => null;

                public void Dispose() { }
            }
        }
```
and change 
```c#
 public SubscriptionManager(IGraphQLExecuter executer, ILoggerFactory loggerFactory)
            : this(executer, loggerFactory, null)
        {
        }
```
to
```c#
 public SubscriptionManager(IGraphQLExecuter executer, ILoggerFactory loggerFactory)
            : this(executer, loggerFactory, NoopFactory.Instance)
        {
        }
```
to keep only one branch?

Reply:
It would be improper to make it a global class because it improperly returns null for the service provider.  So if we did this I would make it private to this class, like this:

```cs
private sealed class NoopServiceScopeFactory : IServiceScopeFactory, IServiceScope
{
    public static IServiceScopeFactory Instance { get; } = new NoopServiceScopeFactory();
    private NoopServiceScopeFactory() { }
    IServiceScope IServiceScopeFactory.CreateScope() => this;
    IServiceProvider IServiceScope.ServiceProvider => null;
    void IDisposable.Dispose() { }
}
```

I probably wouldn't normally do it, but okay...

Reply:
"Global" or not - OK. It was just an example. But why "improperly" ?

Reply:
A service scope should not be returning null for the service provider. If NRT were enabled we would need to force the null return. We do so here only because GraphQL.NET allows null for RequestServices, and only when using the constructor without a factory.